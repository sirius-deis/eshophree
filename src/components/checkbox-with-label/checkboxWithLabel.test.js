import { render, screen, fireEvent } from "@testing-library/react";

import CheckBoxWithLabel from "./checkboxWithLabel.component";

describe("CheckBoxWithLabel", () => {
    const fn = jest.fn();
    it("should render component", () => {
        render(
            <CheckBoxWithLabel checked={true} error="" handler={fn}>
                <p>Some text</p>
            </CheckBoxWithLabel>
        );
        expect(screen.getByText("Some text")).toBeInTheDocument();
    });
    it("should render component which is checked", () => {
        render(
            <CheckBoxWithLabel checked={true} error="" handler={fn}>
                <p>Some text</p>
            </CheckBoxWithLabel>
        );
        expect(screen.getByRole("checkbox")).toBeChecked();
    });
    it("should render component with error", () => {
        render(
            <CheckBoxWithLabel checked={true} error="some error" handler={fn}>
                <p>Some text</p>
            </CheckBoxWithLabel>
        );
        expect(screen.getByText(/some error/)).toBeInTheDocument();
    });
    it("should fire an event on click", () => {
        render(
            <CheckBoxWithLabel checked={true} error="some error" handler={fn}>
                <p>Some text</p>
            </CheckBoxWithLabel>
        );
        fireEvent.click(screen.getByRole("checkbox"));
        expect(fn).toBeCalled();
    });
    it("should fire an event on click and pass false to function", () => {
        render(
            <CheckBoxWithLabel checked={true} error="some error" handler={fn}>
                <p>Some text</p>
            </CheckBoxWithLabel>
        );
        fireEvent.click(screen.getByRole("checkbox"));
        expect(fn).toBeCalledWith(false);
    });
    it("should fire an event on click and pass true to function", () => {
        render(
            <CheckBoxWithLabel checked={false} error="some error" handler={fn}>
                <p>Some text</p>
            </CheckBoxWithLabel>
        );
        fireEvent.click(screen.getByRole("checkbox"));
        expect(fn).toBeCalledWith(true);
    });
});
