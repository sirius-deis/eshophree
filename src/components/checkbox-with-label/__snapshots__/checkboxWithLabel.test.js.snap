// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel component should match snapshot 1`] = `
<div>
  <label
    class="sc-bgqQcB jalWKY"
  >
    Test Checkbox
    <input
      class="sc-gTRrQi rUnwI"
      type="checkbox"
    />
    <span
      class="sc-ewnqHT hxQRMg"
    />
  </label>
</div>
`;
